services/loki.nix

```diff
@@ -0,0 +1,79 @@
+{ config, lib, ... }:
+let
+  dataDir = "/var/lib/loki";
+in {
+
+  services.loki = {
+    inherit dataDir;
+    enable = true;
+    configuration = {
+      auth_enabledi = false;
+      server = {
+        http_listen_port = 3100;
+      };
+      ingester = {
+        lifecycler= {
+          address = "127.0.0.1";
+          ring = {
+            kvstore = {
+              store =  "inmemory";
+            };
+            replication_factor = 1;
+          };
+          final_sleep = "0s";
+        };
+        chunk_idle_period = "5m";
+        chunk_retain_period = "30s";
+      };
+      schema_config = {
+        configs = [{
+          from = "2018-04-15";
+          store = "boltdb";
+          object_store = "filesystem";
+          schema = "v9";
+          index = {
+            prefix = "index_";
+            period = "168h";
+          };
+        }];
+      };
+
+      storage_config = {
+        boltdb = {
+          directory = "/tmp/loki/index";
```

Comment:
do we really want the storage for the service in /tmp?

Reply:
no that was for testing lets get that out of there